src/Streamly/Prelude.hs

```diff
@@ -335,13 +335,10 @@ head m =
 
 -- | Extract the last element of the stream, if any.
 last :: (Streaming t, Monad m) => t m a -> m (Maybe a)
-last m = go (toStream m)
-    where
-    go m1 =
-        let stop            = return Nothing
```

Comment:
@harendra-kumar The problem in this implementation was that stop is always called after the last `yield a Nothing` as a result in the sequence of say `1 .: 2 .: 3`,  `Just 3` is yielded as the last element but finally `Nothing` is also returned. I was not sure how to stop `Nothing` from being called so wrote a new simple implementation. Literally wrote this in a couple of minutes, so please let me know if you have any better suggestions.

Reply:
> The problem in this implementation was that stop is always called after the last yield a Nothing

`stop` cannot be called if `yield a Nothing` is called, because `last` terminates if `yield a Nothing` is called. The problem is that in some cases `stop` is called INSTEAD OF `yield a Nothing`.